### PR TITLE
Enhancement in forms

### PR DESCRIPTION
**Pull request by AbhipsaGuru1012**
## Issue that this pull request solves

 Closes: #488 

## Proposed changes
Enhancement in forms

### Brief description of what is fixed or changed
Added recaptcha in ContactUs form and added validation in ForgotPassword Form

## Types of changes

_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
Hitk

contact-us

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @AbhipsaGuru1012 Please do the following
1. Remove the `node-modules` and `package-lock.json` files from this PR
2. After clicking on `submit` button the website is reloading, please fix it
3. There is no `required` validation in this form, please add it
4. please reduce the bottom-margin of the error message, it should appear closer to the input box

**Comment by AbhipsaGuru1012** [comment]
> Hey @AbhipsaGuru1012 Please do the following
> 
> 1. Remove the `node-modules` and `package-lock.json` files from this PR
> 2. After clicking on `submit` button the website is reloading, please fix it
> 3. There is no `required` validation in this form, please add it
> 4. please reduce the bottom-margin of the error message, it should appear closer to the input box

Okay sure, I will the changes and update soon.

**Comment by AbhipsaGuru1012** [comment]
@Kajol-Kumari I have made the changes. Please review.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari I have made the changes. Please review.

Hey @AbhipsaGuru1012 Please resolve the conflicts here

**Comment by AbhipsaGuru1012** [comment]
> > @Kajol-Kumari I have made the changes. Please review.
> 
> Hey @AbhipsaGuru1012 Please resolve the conflicts here

@Kajol-Kumari I have resolved the conflicts.

File: frontend/src/pages/ContactUs/ContactUs.jsx [hunk 1]
```diff
@@ -115,6 +140,6 @@ export const ContactUs = (props) => {
           </div>
         </div>
       </div>
-    </div>
```
Comment (Kajol-Kumari) [inline frontend/src/pages/ContactUs/ContactUs.jsx:118]: Hey @AbhipsaGuru1012 while resolving the conflict, you left some bug(probably deleted closing div and braces from here) Please fix it. (You would be able to figure out the error just by runing this branch's code)

Comment (AbhipsaGuru1012) [reply frontend/src/pages/ContactUs/ContactUs.jsx]: @Kajol-Kumari I have made some modifications. Please let me know if this is okay.

File: frontend/src/pages/ForgotPassword/ForgotPassword.jsx [hunk 2]
```diff
@@ -79,22 +142,46 @@ export function ForgotPassword() {
...
+                    <div className={`form-group ${style["form-group"]}`}>
+                      <div className={style["forgot-password-input"]}>
+                        <input
+                          autocomplete="off"
+                          id="username"
+                          type="email"
+                          name="email"
+                          placeholder="Email"
+                          onChange={handleChange}
+                          required="required"
```
Comment (Kajol-Kumari) [inline frontend/src/pages/ForgotPassword/ForgotPassword.jsx:161]: Hey @AbhipsaGuru1012 as we don't need any default error message to be displayed anymore so I think we should remove it

File: frontend/src/pages/ForgotPassword/ForgotPassword.jsx [hunk 3]
```diff
@@ -79,22 +142,46 @@ export function ForgotPassword() {
...
+                          className={style["input-forgot-password"]}
+                          onInvalid={(e) => {
+                            e.target.setCustomValidity(
```
Comment (Kajol-Kumari) [inline frontend/src/pages/ForgotPassword/ForgotPassword.jsx:164]: as we are using `joi-validator` so do we need this code snippet from here to [here](https://github.com/HITK-TECH-Community/Community-Website/pull/540/files#diff-9f3cf1c20eff239632332e6daf33e42638fa5cb5bca0ddff9b203afc43c3ef37R175)

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @AbhipsaGuru1012 please incorporate these minor required changes

File: frontend/src/pages/ForgotPassword/ForgotPassword.jsx (hunk 2 above)
Comment (AbhipsaGuru1012) [reply frontend/src/pages/ForgotPassword/ForgotPassword.jsx:161]: @Kajol-Kumari I am working on the latest changes. It is taking a bit of time.

Comment (Kajol-Kumari) [reply frontend/src/pages/ForgotPassword/ForgotPassword.jsx:161]: No worries, please take your time and keep us updated incase you need more than 3-5 days for this PR

**Comment by AbhipsaGuru1012** [comment]
@Kajol-Kumari I have made the required changes. Please let me know if this is okay.

**Review by Kajol-Kumari — APPROVED** [review]
Thanks for incorporating the requested changes @AbhipsaGuru1012 
LGTM :+1:

**Pull request merged**